### ISSUE TITLE
Clean command : Remove S3 bucket and DynamoDB table

### ISSUE THREAD

**Issue by qrevel** [gruntwork-io/terragrunt#250]
Terragrunt is a really nice wrapper to automate the creation of S3 bucket and DynamoDB table.
I'm actually using terragrunt to spawn demo enviroment for different development teams. So each team has access to one or multiple demos environment with similar configuration.
Each demo has its own stack (S3 bucket, DynamoDB table, and Terraform configuration).
Once the demo is finished, I need to clean all the stack.
Unfortunately, I was unable to find any command to clean everything, even the bucket and the table.

Thanks

**Comment by brikis98** [comment]
Hm, I wonder how common of a use case this would be? Usually, Terraform state and locks are important info, so I wouldn't expect it to be deleted too often. On the other hand, for experimenting, learning, and testing, it would be nice to have an easy way to clean up, so I guess I'm not against a `terragrunt-clean` command. PRs welcome :)

**Comment by ryan-dyer-sp** [comment]
Would prefer that the clean only clean specific keys from the bucket related to what was created, not the entire bucket.

**Comment by qrevel** [comment]
Agree with @ryan-dyer-sp for the default clean behavior. Maybe a `--bucket` option to also clean the bucket

**Comment by tomdavidson** [comment]
I want a backend destroy function. The case is for review apps - short lived stage envs that have a 1:1 relationship with a feature branch in trunk-based workflow. Each review app gets its own backend and does not share a backend with other envs. 

br, tom

**Comment by nilsmeyer** [comment]
One possible issue with this is that you can use the same bucket and dynamodb table for multiple terraform states. You can make this safe by checking whether the bucket is empty or you just trust the user to know what they're doing - possibly with a `--yes-i-know-what-i-am-doing-nuke-it-please` flag.

**Comment by Tensho** [comment]
Have the same case for short-lived preview apps.

**Comment by evya123** [comment]
Anything new about this feature? Would be very handy when experimenting

**Comment by sergibarroso-datarobot** [comment]
+1

**Comment by mjozefcz** [comment]
+1

**Comment by YKolokoltsev** [comment]
+1

**Comment by yhakbar** [comment]
We're adding a dedicated `backend` command as part of the CLI Redesign in https://github.com/gruntwork-io/terragrunt/issues/3445.

Currently, the RFC only explores adding the ability to explicitly flush out state related to a particular unit with the following:

```bash
terragrunt backend delete
```

But we could theoretically have something like:

```bash
terragrunt backend explode
```

or

```bash
terragrunt backend nuke
```

The risks involved with programmatically destroying backends is non-trivial, so we'll want to be careful when introducing this capability.

**Comment by aslafy-z** [comment]
A safer alternative could be a `terragrunt backend relocate` that would first retrieve the state locally, then remove the remote state.
This way, it is the responsibility of the operator to `rm -f` the tfstate.

**Comment by ajschmidt8** [comment]
Not stale
